code/messaging/src/phone/java/com/adobe/marketing/mobile/MessagingDefaultImageDownloader.java

```diff
@@ -0,0 +1,74 @@
+package com.adobe.marketing.mobile;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.os.AsyncTask;
+import android.util.Log;
+import android.util.LruCache;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+class MessagingDefaultImageDownloader implements IMessagingImageDownloader {
+    private final LruCache<String, Bitmap> cache;
+    private final ExecutorService executorService;
+
+    private static volatile MessagingDefaultImageDownloader singletonInstance = null;
+
+    public static MessagingDefaultImageDownloader getInstance() {
```

Comment:
Let's use the `Bill Pugh Singleton Implementation`. It is the most popular way to create singletons.
https://www.geeksforgeeks.org/java-singleton-design-pattern-practices-examples/